src/plugins/kibana/public/discover/controllers/discover.js

```diff
@@ -217,7 +217,10 @@ define(function (require) {
             if (rows == null && oldRows == null) return status.LOADING;
 
             var rowsEmpty = _.isEmpty(rows);
-            if (rowsEmpty && fetchStatus) return status.LOADING;
+            // an undefined fetchStatus means the requests are still being
+            // prepared to be sent
+            if (_.isUndefined(fetchStatus)) return status.LOADING;
+            else if (rowsEmpty && fetchStatus) return status.LOADING;
```

Comment:
Hmm, eh, If that's the case I would say I don't really care if you move them. Perhaps a comment in the code though about it being null possibly too.